### PR TITLE
URI encode url query parameter values on prefetch templates from CDS Services

### PR DESCRIPTION
**Pull request by zplata**
Issue: https://github.com/cds-hooks/sandbox/issues/64

Currently, CDS Services can define a prefetch request in their CDS Service definition. This mainly allows services to prefetch data from the FHIR server leveraged by the EHR (Sandbox) if allowable. The Sandbox will only simply fill in the relevant Patient.Id in context for prefetch templates and then execute a request to the FHIR server in context for the requested CDS Service data. However, the Sandbox does not encode URIs on additional query parameters tagged onto a prefetch request. This causes an issue as in the issue above, where specified `code` parameters will fail the call due to inclusion of the system (i.e. `http://loinc.org`). 

The Sandbox should inject the patient ID per usual into the prefetch template, and additionally use the native `encodeURIComponent` to encode the query parameter values in the rest of the template if necessary.

@kpshek 
@mjhenkes 
@kolkheang

### CODE REVIEW

File: src/scripts/stores/DecisionStore.js [hunk 1]
```diff
@@ -34,11 +34,50 @@ function getFhirContext() {
...
+    params[p[0]] = p[1];
+  });
+
+  return params;
+}
+
+function constructURLParams(params) {
+  var urlQuery = '';
+  var i = 0;
+  Object.keys(params).forEach(function(param) {
```
Comment (kolkheang) [inline src/scripts/stores/DecisionStore.js]: I'd name `param` as `key` instead.

File: src/scripts/stores/DecisionStore.js [hunk 2]
```diff
@@ -34,11 +34,50 @@ function getFhirContext() {
...
+  return urlQuery;
+}
+
 function fillTemplate(template, context) {
-  var flat = JSON.stringify(template)
-  .replace(/{{\s*Patient\.id\s*}}/g, context["Patient.id"])
-  return JSON.parse(flat)
+  var flat = template.replace(/{{\s*Patient\.id\s*}}/g, context["Patient.id"]).toString();
+  if (flat.split('?').length > 1) {
+    var splitUrl = flat.split('?');
```
Comment (kolkheang) [inline src/scripts/stores/DecisionStore.js]: I'd add a comment here to indicate that if query param isn't URI component encoded then encode each component

Comment (zplata) [reply src/scripts/stores/DecisionStore.js]: addressed in https://github.com/cerner/cds-hooks-sandbox/pull/36/commits/498984d3dab3e7a6461da64daa121e71bb4263e8

File: src/scripts/stores/DecisionStore.js [hunk 3]
```diff
@@ -34,11 +34,50 @@ function getFhirContext() {
   return c.set('Patient.id', c.get('patient')).toJS()
 }
 
+function getURLParams(url) {
```
[the remark this answers is not in the document]
Comment (zplata) [reply src/scripts/stores/DecisionStore.js]: addressed in https://github.com/cerner/cds-hooks-sandbox/pull/36/commits/498984d3dab3e7a6461da64daa121e71bb4263e8

**Pull request merged**